django_cas_ng/migrations/0001_initial.py

```diff
@@ -15,26 +15,25 @@ class Migration(migrations.Migration):
 
     operations = [
         migrations.CreateModel(
-            name='ProxyGrantingTicket',
+            name='SessionTicket',
             fields=[
                 ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
-                ('session_key', models.CharField(blank=True, max_length=255, null=True)),
-                ('pgtiou', models.CharField(blank=True, max_length=255, null=True)),
-                ('pgt', models.CharField(blank=True, max_length=255, null=True)),
-                ('date', models.DateTimeField(auto_now_add=True)),
-                ('user', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='+', to=settings.AUTH_USER_MODEL)),
+                ('session_key', models.CharField(max_length=1024)),
+                ('ticket', models.CharField(max_length=1024)),
             ],
         ),
         migrations.CreateModel(
-            name='SessionTicket',
+            name='ProxyGrantingTicket',
             fields=[
                 ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
-                ('session_key', models.CharField(max_length=255)),
-                ('ticket', models.CharField(max_length=255)),
+                ('session_key', models.CharField(blank=True, max_length=1024, null=True)),
```

Comment:
I think the root cause the char length too long caused mysql can not support it in index. Maybe we should reduce it to a lower number like 760 (any number smaller than 767).

Quote from https://dev.mysql.com/doc/refman/8.0/en/innodb-limits.html:


> The index key prefix length limit is 767 bytes for InnoDB tables that use the [REDUNDANT](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_redundant_row_format) or [COMPACT](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_compact_row_format) row format. For example, you might hit this limit with a [column prefix](https://dev.mysql.com/doc/refman/8.0/en/glossary.html#glos_column_prefix) index of more than 191 characters on a TEXT or VARCHAR column, assuming a utf8mb4 character set and the maximum of 4 bytes for each character.

Reply:
Yeah - if you want to increase the max_length of a database field, that should happen in a new migration file, not by altering the existing one.

Reply:
I see. never tried it before.

Reply:
the problem is, django_cas_ng.0002_auto_20201023_1400 causes an exception so subsequent migrations will not be applied,